aws/resource_aws_api_gateway_base_path_mapping.go

```diff
@@ -87,6 +88,69 @@ func resourceAwsApiGatewayBasePathMappingCreate(d *schema.ResourceData, meta int
 	return resourceAwsApiGatewayBasePathMappingRead(d, meta)
 }
 
+func resourceAwsApiGatewayBasePathMappingUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).apigatewayconn
+
+	operations := make([]*apigateway.PatchOperation, 0)
+
+	if d.HasChange("stage_name") {
+		operations = append(operations, &apigateway.PatchOperation{
+			Op:    aws.String("replace"),
+			Path:  aws.String("/stage"),
+			Value: aws.String(d.Get("stage_name").(string)),
+		})
+	}
+
+	if d.HasChange("api_id") {
+		operations = append(operations, &apigateway.PatchOperation{
+			Op:    aws.String("replace"),
+			Path:  aws.String("/restapiId"),
+			Value: aws.String(d.Get("api_id").(string)),
+		})
+	}
+
+	if d.HasChange("base_path") {
+		operations = append(operations, &apigateway.PatchOperation{
+			Op:    aws.String("replace"),
+			Path:  aws.String("/basePath"),
+			Value: aws.String(d.Get("base_path").(string)),
+		})
+	}
+
+	domainName, basePath, decodeErr := decodeApiGatewayBasePathMappingId(d.Id())
+	if decodeErr != nil {
+		return decodeErr
+	}
+
+	input := apigateway.UpdateBasePathMappingInput{
+		BasePath:        aws.String(basePath),
+		DomainName:      aws.String(domainName),
+		PatchOperations: operations,
+	}
+
+	log.Printf("[INFO] Updating API Gateway base path mapping: %s", input)
+
+	_, err := conn.UpdateBasePathMapping(&input)
+
+	if err != nil {
+		if isResourceTimeoutError(err) {
+			_, err = conn.UpdateBasePathMapping(&input)
+		}
```

Comment:
Nit: Logic is not using `resource.Retry()` logic so `isResourceTimeoutError(err)` check is extraneous.

```suggestion
```